### ISSUE TITLE
getting a step-by-step tibble summary from prepped recipe

### ISSUE THREAD

**Issue by kylegilde** [tidymodels/recipes#604]
Hi,

When using the new 0.1.15 logical expressions to select columns, I was wondering if there is an easy, programmatic way to get a tibble/data.frame of the columns that were selected/altered & removed/added for each step from a prepped recipe.

Thank you!

**Comment by topepo** [comment]
As of the previous version of `recipes`... [yes](https://www.tidyverse.org/blog/2020/10/recipes-0-1-14/#other-changes)

**Comment by kylegilde** [comment]
Yes, this is what I'm looking for.

Is there a way to make the recipes print without truncating("...") a long list of column names?

Thanks!

**Comment by topepo** [comment]
Not really.

This is a horrible, inelegant, and inefficient method: 

``` r
library(tidymodels)
#> ── Attaching packages ───────────────────────────────── tidymodels 0.1.1.9000 ──
#> ✓ broom     0.7.2          ✓ recipes   0.1.15    
#> ✓ dials     0.0.9          ✓ rsample   0.0.8     
#> ✓ dplyr     1.0.2          ✓ tibble    3.0.4     
#> ✓ ggplot2   3.3.2          ✓ tidyr     1.1.2     
#> ✓ infer     0.5.3          ✓ tune      0.1.1.9001
#> ✓ modeldata 0.1.0          ✓ workflows 0.2.1     
#> ✓ parsnip   0.1.4.9000     ✓ yardstick 0.0.7     
#> ✓ purrr     0.3.4
#> ── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
#> x purrr::discard() masks scales::discard()
#> x dplyr::filter()  masks stats::filter()
#> x dplyr::lag()     masks stats::lag()
#> x recipes::step()  masks stats::step()
data(ames)
ames$Sale_Price <- log10(ames$Sale_Price)

ames_rec <- recipe(Sale_Price ~ ., data = ames) %>%
  step_BoxCox(Lot_Area, Gr_Liv_Area) %>%
  step_other(Neighborhood, threshold = 0.05)  %>%
  step_dummy(all_nominal()) %>%
  step_interact(~ starts_with("Central_Air"):Year_Built) %>%
  step_ns(Longitude, Latitude, deg_free = 5)


get_names <- function(i, x) {
  new_rec <- x
  new_rec$steps <- x$steps[1:i]
  new_rec <- prep(new_rec)
  names(juice(new_rec))
}

prep_print <- function(rec) {
  n_steps <- length(rec$step)
  map(1:n_steps, ~ get_names(.x, rec))
}

nms <- prep_print(ames_rec)
str(nms)
#> List of 5
#>  $ : chr [1:74] "MS_SubClass" "MS_Zoning" "Lot_Frontage" "Lot_Area" ...
#>  $ : chr [1:74] "MS_SubClass" "MS_Zoning" "Lot_Frontage" "Lot_Area" ...
#>  $ : chr [1:257] "Lot_Frontage" "Lot_Area" "Year_Built" "Year_Remod_Add" ...
#>  $ : chr [1:258] "Lot_Frontage" "Lot_Area" "Year_Built" "Year_Remod_Add" ...
#>  $ : chr [1:266] "Lot_Frontage" "Lot_Area" "Year_Built" "Year_Remod_Add" ...
```

<sup>Created on 2020-11-12 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by kylegilde** [comment]
In your example, step_BoxCox is only given 2 variables, but the list you created with your functions is showing 74?

**Comment by topepo** [comment]
The vectors are the current columns in the data set. There are 74 columns in `ames`

**Comment by kylegilde** [comment]
I see, so your functions give me the cumulative column names after each step. This does help.

 Is there a way to extract the input column names to each step, especially when the selection mechanism is a logical expression.

**Comment by juliasilge** [comment]
Our approach so far has been to expose that kind of information in logging, as Max linked to earlier in the thread. We haven't worked to expose this kind of info programmatically per step, because of efficiency and practical reasons. Instead, the selection is just executed. Certainly you can `prep()`/`bake()` a recipe step-by-step during interactive debugging, and we see the logging option as filling a need for non-interactive logging.

Can you explain a little more about your use case and motivation, so we can weigh that against other concerns?

**Comment by kylegilde** [comment]
My goal is to be able to explain what happened in the recipe and to be able to retrieve that at a latter point. I want to be able to get a list of the input, output & removed features per step. 

Currently, the logging truncates the list of features, so it's not really providing a full picture of what happened.

It would be nice if this info was programmatically accessible, but I have hacked my way to a solution. One can get a list of removed features out of the feature filtering steps, and I have been able to map input to output features for the most part: https://github.com/tidymodels/recipes/issues/611
